google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Publisher.java

```diff
@@ -287,7 +292,44 @@ public void run() {
           });
     }
 
-    return publishResult;
+    return new ListenableFutureDelegate<String>(publishResult);
+  }
+
+  private static class ListenableFutureDelegate<V> extends SimpleForwardingListenableFuture<V>
```

Comment:
No, this can only be copied. It is true that `ListenableFutureDelegate` class is also in GAX, but we cannot make the class public. If it were public, it becomes a part of GAX's surface and we defeat the purpose of not exposing `ListenableFuture`.

Unfortunately we'll just have to copy this to every package that needs it.